android/app/src/main/java/org/chrome/device/ml/experiments/MobileBertExperiment.java

```diff
@@ -0,0 +1,115 @@
+/* Copyright 2020. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+package org.chrome.device.ml.experiments;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.Message;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.chrome.device.ml.thirdparty.ml.LoadDatasetClient;
+import org.chrome.device.ml.thirdparty.ml.QaAnswer;
+import org.chrome.device.ml.thirdparty.ml.QaClient;
+
+/**
+ * Mobilebert Question/Answer Experiment
+ */
+public class MobileBertExperiment implements Experiment {
+    private static final String TAG = "MobileBertExperiment";
+    private static final String MODEL_PATH = "bert_model.tflite";
+
+    private final Context context;
+    private Handler handler;
+    private Handler UIHandler;
+
+    private LoadDatasetClient datasetClient;
+    private QaClient qaClient;
+    private ArrayList<Double> timing;
+
+    public MobileBertExperiment(Context context, Handler handler) {
+        this.context = context;
+        this.UIHandler = handler;
+
+        this.datasetClient = new LoadDatasetClient(this.context);
+
+        HandlerThread handlerThread = new HandlerThread("BertExp");
+        handlerThread.start();
+
+        this.handler = new Handler(handlerThread.getLooper());
+        this.qaClient = new QaClient(this.context, MODEL_PATH);
+        this.timing = new ArrayList<Double>();
+    }
+
+    public void initialize() {
+        handler.post(
+                () -> {
+                    qaClient.loadModel();
+                    qaClient.loadDictionary();
+                });
+    }
+
+    public void close() {
+        handler.post(
+                () -> {
+                    qaClient.close();
+                });
+    }
+
+    // Evaluates Bert model with contents and questions
+    public void evaluate(int numberOfContents) {
+        timing.clear();
```

Comment:
is there any guarantee of the thread safety here? what is getTime() called on. looks like evaluate() is called on UI thread but then the handler thread is modifying it? would be good to add comments regarding semantics and potentially guards around it if it is unsafe

Reply:
getTime() is/should be called after the evaluation is finished, I added comment.

Reply:
id probably feel more comfortable with this if you had some sort of lock around timing for robustness. you cant actually guarantee that this will be called after processing has ended.

Reply:
@sophiechang makes sense. I made both **evaluation()** and **getTime()** synchronized for thread safety.

Reply:
this doesn't work as once you post it to the handler and this method finishes, you are out of that block. it's probably best to have some sort of mutex that guards the timing variable. 

see https://developer.android.com/reference/androidx/annotation/GuardedBy